src/App.tsx

```diff
@@ -8,39 +8,50 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { getUsers } from './api/users';
+import { setUsers } from './features/users/usersSlice';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { selectAuthor, setAuthor } from './features/author/authorSlice';
+import {
+  fetchPostsByUserId,
+  selectPosts,
+  selectPostsStatus,
+} from './features/posts/postsSlice';
+import {
+  selectSelectedPost,
+  setSelectedPost,
+} from './features/selectedPost/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const author = useAppSelector(selectAuthor);
+  const posts = useAppSelector(selectPosts);
+  const postsStatus = useAppSelector(selectPostsStatus);
+  const selectedPost = useAppSelector(selectSelectedPost);
+  const dispatch = useAppDispatch();
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  useEffect(() => {
+    getUsers().then(users => dispatch(setUsers(users)));
+  }, [dispatch]);
 
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
+  // function loadUserPosts(userId: number) {
+  //   setLoaded(false);
 
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  //   getUserPosts(userId)
+  //     .then(items => dispatch(setPosts(items)))
+  //     .catch(() => dispatch(setError(true)))
+  //     // We disable the spinner in any case
+  //     .finally(() => setLoaded(true));
+  // }
```

Comment:
pls remove redundant comments

Reply:
thank you, will fix! Just for info, I checked demo version in the task description, and it is possible to submit with white spaces there as well